cmd/saga/internal/auth/auth.go

```diff
@@ -0,0 +1,128 @@
+package auth
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/adrg/xdg"
+	"github.com/spf13/cobra"
+	"go.einride.tech/aip-cli/aipcli"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+)
+
+const (
+	authConfigFile = "saga/auth.json"
+)
+
+func Authenticate(host string, verboseLogging bool) (string, error) {
```

Comment:
This should probably take a context and use it in the networking calls below.
```suggestion
func Authenticate(ctx context.Context, host string, verboseLogging bool) (string, error) {
```